File: 0001-two-sum/0001-two-sum.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> dict;
        int val = 0;

        for (int i = 0; i < nums.size(); i++) {
            dict[nums[i]] = i;
        }

        for (int i = 0; i < nums.size(); i++) {
            val = target - nums[i];
            if (dict.find(val) != dict.end()) {
                if (dict[val] != i) {
                    return {i, dict[val]};
                }
            }
        }

        return {0,0};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //int[] myNums = nums;
        int listLen = nums.length;
        for (int i = 1; i < listLen; i++) {
            for (int j = i; j < listLen; j++) {
                if (target == (nums[j] + nums[j-i])) {
                    return new int[] {j, j-i};
                }

            }
        }
        return null;
    }
}